File: app/java/src/main/java/com/example/production/domain/model/inventory/StockAdjustment.java
package com.example.production.domain.model.inventory;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 在庫調整データエンティティ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StockAdjustment {
    private Long id;
    private String adjustmentNumber;
    private String stocktakingNumber;
    private String itemCode;
    private String locationCode;
    private LocalDate adjustmentDate;
    private String adjusterCode;
    private BigDecimal adjustmentQuantity;
    private String reasonCode;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Inventory/StockAdjustment.cs
namespace ProductionManagement.Domain.Models.Inventory;

/// <summary>
/// 在庫調整データ
/// </summary>
public class StockAdjustment
{
    public long? Id { get; init; }
    public required string AdjustmentNumber { get; init; }
    public string? StocktakingNumber { get; init; }
    public required string ItemCode { get; init; }
    public required string LocationCode { get; init; }
    public required DateOnly AdjustmentDate { get; init; }
    public required string AdjusterCode { get; init; }
    public required decimal AdjustmentQuantity { get; init; }
    public required string ReasonCode { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }
}
